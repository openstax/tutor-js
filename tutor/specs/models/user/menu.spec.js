import SnapShot from 'react-test-renderer';
import Courses from '../../../src/models/courses-map';
import UserMenu from '../../../src/models/user/menu';
import User from '../../../src/models/user';
import { bootstrapCoursesList } from '../../courses-test-data';

jest.mock('../../../src/models/user', () => ({
  isConfirmedFaculty: true,
}));

describe('Current User Store', function() {

  beforeEach(function() {
    bootstrapCoursesList();
  });

  afterEach(function() {
    Courses.clear();
  });

  it('computes help URL', () => {
    expect(UserMenu.helpURL).toContain('help');
  });

  it('should return expected menu routes for courses', function() {
    User.isConfirmedFaculty = false;
    expect(UserMenu.getRoutes('1')).toMatchSnapshot();
    expect(UserMenu.getRoutes('2')).toMatchSnapshot();
    Courses.clear();
    expect(UserMenu.getRoutes()).toMatchSnapshot();
  });
});
