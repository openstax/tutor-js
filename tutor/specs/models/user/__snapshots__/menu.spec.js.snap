// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current User Store should return expected menu routes for a student 1`] = `
Array [
  Object {
    "label": "My Courses",
    "name": "myCourses",
    "options": Object {
      "separator": "after",
    },
  },
  Object {
    "label": "Dashboard",
    "name": "dashboard",
    "options": Object {},
    "params": Object {
      "courseId": 2,
    },
  },
  Object {
    "label": "Browse the Book",
    "name": "browseBook",
    "options": Object {},
    "params": Object {
      "courseId": 2,
    },
  },
  Object {
    "label": "Scores",
    "name": "viewScores",
    "options": Object {},
    "params": Object {
      "courseId": 2,
    },
  },
  Object {
    "label": "Performance Forecast",
    "name": "viewPerformanceGuide",
    "options": Object {},
    "params": Object {
      "courseId": 2,
    },
  },
  Object {
    "label": "Change Student ID",
    "name": "changeStudentId",
    "options": Object {},
    "params": Object {
      "courseId": 2,
    },
  },
  Object {
    "label": "Create a Course",
    "name": "createNewCourse",
    "options": Object {
      "separator": "both",
    },
    "params": Object {
      "courseId": 2,
    },
  },
]
`;

exports[`Current User Store should return expected menu routes for a teacher 1`] = `
Array [
  Object {
    "label": "My Courses",
    "name": "myCourses",
    "options": Object {
      "separator": "after",
    },
  },
  Object {
    "label": "Dashboard",
    "name": "dashboard",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Browse the Book",
    "name": "browseBook",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Performance Forecast",
    "name": "viewPerformanceGuide",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Question Library",
    "name": "viewQuestionsLibrary",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Student Scores",
    "name": "viewScores",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Course Settings",
    "name": "courseSettings",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Course Roster",
    "name": "courseRoster",
    "options": Object {},
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Create a Course",
    "name": "createNewCourse",
    "options": Object {
      "separator": "both",
    },
    "params": Object {
      "courseId": 1,
    },
  },
  Object {
    "label": "Copy this Course",
    "name": "createNewCourse",
    "options": Object {
      "key": "cloneCourse",
      "separator": "after",
    },
    "params": Object {
      "sourceId": 1,
    },
  },
]
`;

exports[`Current User Store should return expected menu routes when course is missing 1`] = `
Array [
  Object {
    "label": "Create a Course",
    "name": "createNewCourse",
    "options": Object {
      "separator": "both",
    },
  },
]
`;
